src/filesystem.cc

```diff
@@ -505,16 +540,17 @@ class GCSFileSystem : public FileSystem {
   google::cloud::StatusOr<gcs::Client> client_;
 };
 
-GCSFileSystem::GCSFileSystem()
+GCSFileSystem::GCSFileSystem(const GCSCredential& gs_cred)
 {
-  client_ = gcs::Client::CreateDefaultClient();
+  auto creds = gcs::oauth2::CreateServiceAccountCredentialsFromJsonFilePath(gs_cred.path_);
+  if (creds) {
```

Comment:
what happens for invalid creds?

Reply:
The `FileSystemManager` will always call `CheckClient` before returning the file system. If the check fails, it reloads the credential and retry. If the check also fails after reload, the error from `CheckClient` is returned and the server gracefully terminates. It will print something like "Unable to create GCS client. Check account credentials."